games-fps/chocolate-doom/chocolate_doom-3.0.0.recipe

```diff
@@ -0,0 +1,128 @@
+DESCRIPTION="Chocolate Doom is a source port of the game Doom, by id Software. \
+The purpose of Chocolate Doom is to be as compatible as possible with the \
+original DOS version of Doom."
+HOMEPAGE="https://www.chocolate-doom.org/"
+COPYRIGHT="1993-1996, Id Software, Inc
+	1993-2008, Raven Software
+	1996, Rogue Entertainment / Velocity, Inc.
+	1996-2013, Free Software Foundation, Inc
+	1999, Thomas A. Fine
+	2002-2003, Marcel Telka
+	2002-2010, The DOSBox Team
+	2005-2014, Simon Howard
+	2005-2014, Andrey Budko
+	2006, Ben Ryves
+	2008, GhostlyDeath
+	2008, David Flater
+	2010, James Haley, Samuel Villarreal
+	2011, Team Eternity
+	2014, Fabian Greffrath"
+LICENSE="GNU GPL v2
+	BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://www.chocolate-doom.org/downloads/$portVersion/chocolate-doom-$portVersion.tar.gz"
+CHECKSUM_SHA256="73aea623930c7d18a7a778eea391e1ddfbe90ad1ac40a91b380afca4b0e1dab8"
+SOURCE_DIR="chocolate-doom-$portVersion"
+ADDITIONAL_FILES="
+	chocolate_doom.rdef.in
+	chocolate_doom.png
+	"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	chocolate_doom$commandSuffix = $portVersion
+	cmd:chocolate_doom$commandSuffix = $portVersion
+	cmd:chocolate_doom_setup$commandSuffix = $portVersion
+	cmd:chocolate_heretic_setup$commandSuffix = $portVersion
+	cmd:chocolate_heretic$commandSuffix = $portVersion
+	cmd:chocolate_hexen_setup$commandSuffix = $portVersion
+	cmd:chocolate_hexen$commandSuffix = $portVersion
+	cmd:chocolate_server$commandSuffix = $portVersion
+	cmd:chocolate_setup$commandSuffix = $portVersion
+	cmd:chocolate_strife_setup$commandSuffix = $portVersion
+	cmd:chocolate_strife$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	lib:libSDL2_mixer_2.0$secondaryArchSuffix
+	lib:libSDL2_net_2.0$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libsdl2_mixer$secondaryArchSuffix
+	devel:libsdl2_net$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:install
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python
+	cmd:sed
+	"
+
+defineDebugInfoPackage chocolate_doom \
+	$commandBinDir/chocolate-doom$commandSuffix \
+	$commandBinDir/chocolate-doom-setup$commandSuffix \
+	$commandBinDir/chocolate-heretic-setup$commandSuffix \
+	$commandBinDir/chocolate-heretic$commandSuffix \
+	$commandBinDir/chocolate-hexen-setup$commandSuffix \
+	$commandBinDir/chocolate-hexen$commandSuffix \
+	$commandBinDir/chocolate-server$commandSuffix \
+	$commandBinDir/chocolate-setup$commandSuffix \
+	$commandBinDir/chocolate-strife-setup$commandSuffix \
+	$commandBinDir/chocolate-strife$commandSuffix
```

Comment:
There is a small **_latent_** error here, which does not get noticed in Haiku x86_gcc2 hybrid (i.e. with x86_gcc2 primary arch with x86 secondary arch), but which **_would_** be noticed in custom hybrid builds of Haiku such as x86_64 primary arch with x86 secondary arch.

```diff
-defineDebugInfoPackage chocolate_doom \
+defineDebugInfoPackage chocolate_doom$secondaryArchSuffix \
-	$commandBinDir/chocolate-doom$commandSuffix \
+	$commandBinDir/chocolate-doom \
-	$commandBinDir/chocolate-doom-setup$commandSuffix \
+	$commandBinDir/chocolate-doom-setup \
-	$commandBinDir/chocolate-heretic-setup$commandSuffix \
+	$commandBinDir/chocolate-heretic-setup \
-	$commandBinDir/chocolate-heretic$commandSuffix \
+	$commandBinDir/chocolate-heretic \
-	$commandBinDir/chocolate-hexen-setup$commandSuffix \
+	$commandBinDir/chocolate-hexen-setup \
-	$commandBinDir/chocolate-hexen$commandSuffix \
+	$commandBinDir/chocolate-hexen \
-	$commandBinDir/chocolate-server$commandSuffix \
+	$commandBinDir/chocolate-server \
-	$commandBinDir/chocolate-setup$commandSuffix \
+	$commandBinDir/chocolate-setup \
-	$commandBinDir/chocolate-strife-setup$commandSuffix \
+	$commandBinDir/chocolate-strife-setup \
-	$commandBinDir/chocolate-strife$commandSuffix
+	$commandBinDir/chocolate-strife
```

Explanation: We should not use:
* `$secondaryArchSuffix` in the lines where `$binDir` appears, nor
* `$commandSuffix` in the lines where `$commandBinDir` appears.

This is because :
1) without the commandBinDir/commandSuffix, the runtimes we have in `$binDir` do not end with any suffix.
2) with the commandBinDir/commandSuffix, the runtimes we have in `$commandBinDir` (i.e. `$prefix/bin`) do not end with any suffix.

Reply:
BTW, the suffix is unknown to the build system.
What haikuporter does, when we build a recipe with the defaut `$binDir` (i.e. without using a `commandBinDir`) for a secondary arch, say, x86, is that it creates in `$prefix/bin` a symlink for each runtime found in `$binDir` (i.e. `$prefix/bin/x86/`) after the build.
So, if the build creates `$prefix/bin/x86/foobar` then haikuporter will create in `$prefix/bin/` a symlink pointing to `x86/foobar` and will name that symlink `foobar-x86`.
That's why there is no suffix in any runtime found in `$prefix/bin/x86/`.

Suffixes **_only_** exist in `$prefix/bin/` for symlinks created **_by_** haikuporter **_itself_**.
When we use the commandBinDir/commandSuffix "trick" in a recipe which does not build for the x86_gcc2 primary arch, we have no suffixes at all.

But if someone builds a custom hybrid Haiku, with e.g. x86_64 as primary arch and x86 as secondary arch, then haikuporter would accept to build chocolate_doom_x86.
In such an environment, `BUILD()` and `INSTALL()` would both run correctly, but when haikuporter would try to create the debuginfo package, it would not find `$prefix/bin/x86/chocolate-doom_x86` (nor `$prefix/bin/x86/chocolate-doom-x86`).
As a matter of fact, what we would get would be the runtimes (e.g. `chocolate-doom`) in `$prefix/bin/x86/` and the symlinks (e.g. `chocolate-doom-x86 -> x86/chocolate-doom`) in `$prefix/bin/`.

So if you wish to create a debuginfo package for chocolate_doom then `defineDebugInfoPackage` requires:
* `chocolate_doom$secondaryArchSuffix` for its very first parameter,
* `$commandBinDir/chocolate-doom`, `$commandBinDir/chocolate-doom-setup`, and so on, for each of the runtimes created.